python/ray/data/preprocessors/imputer.py

```diff
@@ -153,8 +154,18 @@ def _transform_pandas(self, df: pd.DataFrame):
             else:
                 if is_categorical_dtype(df.dtypes[column]):
                     df[output_column] = df[column].cat.add_categories([value])
-                if output_column != column:
+
+                if (
+                    output_column != column
+                    # If the backing array is memory-mapped from shared memory, then the
+                    # array won't be writeable.
+                    or (
+                        isinstance(df[output_column].values, np.ndarray)
+                        and not df[output_column].values.flags.writeable
+                    )
+                ):
```

Comment:
The previous implementation assumed that the array is writeable, and this assumptions was only true because we previously batched the map inputs